Mesh_3/include/CGAL/Mesh_3/generate_label_weights.h

```diff
@@ -0,0 +1,267 @@
+  {
+    return p != 255;
+  };
+  auto size = itk_img->GetLargestPossibleRegion().GetSize();
+  return std::count_if(itk_img->GetBufferPointer(),
+                       itk_img->GetBufferPointer() + size[0]*size[1]*size[2],
+                       diff255);
+}
+#endif //CGAL_MESH_3_WEIGHTED_IMAGES_DEBUG
+
+template<typename Image_word_type>
+WORD_KIND get_wordkind()
+{
+  if (std::is_floating_point<Image_word_type>::value)
+    return WK_FLOAT;
+  else
+    return WK_FIXED;
+/** unknown (uninitialized) */
+//    WK_UNKNOWN
+}
+
+template<typename Image_word_type>
+SIGN get_sign()
+{
+  if (std::is_signed<Image_word_type>::value)
+    return SGN_SIGNED;
+  else
+    return SGN_UNSIGNED;
+/** unknown (uninitialized or floating point words) */
+//    SGN_UNKNOWN
+}
+
+}//namespace internal
+
+/// @cond INTERNAL
+template<typename Image_word_type>
+CGAL::Image_3 generate_label_weights_with_known_word_type(const CGAL::Image_3& image,
+                                                    const float& sigma)
+{
+  typedef unsigned char Weights_type; //from 0 t 255
+  const std::size_t img_size = image.size();
+
+  //create weights image
+  _image* weights
+    = _createImage(image.xdim(), image.ydim(), image.zdim(),
+                   1,                                        //vectorial dimension
+                   image.vx(), image.vy(), image.vz(),
+                   sizeof(Weights_type),                     //image word size in bytes
+                   internal::get_wordkind<Weights_type>(),   //image word kind WK_FIXED, WK_FLOAT, WK_UNKNOWN
+                   internal::get_sign<Weights_type>());      //image word sign
+  Weights_type* weights_ptr = (Weights_type*)(weights->data);
+  std::fill(weights_ptr,
+            weights_ptr + img_size,
+            Weights_type(0));
+  weights->tx = image.tx();
+  weights->ty = image.ty();
+  weights->tz = image.tz();
+
+  //convert image to itkImage
+  using ImageType = itk::Image<Image_word_type, 3/*Dimension*/>;
+  using WeightsType = itk::Image<Weights_type, 3>;
+  typename ImageType::Pointer itk_img = ImageType::New();
+  std::set<Image_word_type> labels;
+  internal::convert_image_3_to_itk(image, itk_img.GetPointer(), labels);
+
+  using DuplicatorType = itk::ImageDuplicator<ImageType>;
+  using IndicatorFilter = itk::BinaryThresholdImageFilter<ImageType, WeightsType>;
+  using GaussianFilterType = itk::RecursiveGaussianImageFilter<WeightsType, WeightsType>;
+  using MaximumImageFilterType = itk::MaximumImageFilter<WeightsType>;
+
+  std::vector<typename ImageType::Pointer> indicators(labels.size());
+  typename DuplicatorType::Pointer duplicator = DuplicatorType::New();
+  duplicator->SetInputImage(itk_img);
+  duplicator->Update();
+
+  int id = 0;
+  for (Image_word_type label : labels)
+  {
+    if (id > 0)
+    {
+      duplicator->SetInputImage(indicators[id - 1]);
+      duplicator->Update();
+    }
+    indicators[id++] = duplicator->GetOutput();
+  }
```

Comment:
The loop variable `label` is not used. This loop should be a regular loop:
```suggestion
  
  for (int id = 0, end = labels.size(); id < end; ++id)
  {
    if (id > 0)
    {
      duplicator->SetInputImage(indicators[id - 1]);
      duplicator->Update();
    }
    indicators[id] = duplicator->GetOutput();
  }
```